Slim/Schema.pm

```diff
@@ -241,12 +378,29 @@ sub init {
 	}
 
 	if ( !main::SCANNER ) {
+		require Slim::Control::Request;
 		# Wipe cached data after rescan
 		Slim::Control::Request::subscribe( sub {
 			$class->wipeCaches;
 			Slim::Schema::Album->addReleaseTypeStrings;
 		}, [['rescan'], ['done']] );
 	}
+	elsif ( main::DEBUGLOG ) {
+		require Slim::Control::Request;
+		# Initialize scan-local caches.
+		$class->_initScanContributorCache;
+		$class->_registerScanContributorCacheEndHook;
+
+		# Scanner-only stats, emitted once per scan if DEBUGLOG is enabled.
+		Slim::Control::Request::subscribe( sub {
+			$class->_logScanContributorCacheStats;
+		}, [['rescan'], ['done']] );
+	}
+	elsif ( main::SCANNER ) {
+		require Slim::Control::Request;
+		# Initialize scan-local caches even if DEBUGLOG is off.
+		$class->_initScanContributorCache;
+	}
```

Comment:
That's not going to fly. We don't want to use `Slim::Control::Request` in the scanner. You don't need to subscribe to those events in the scanner, do you? The scanner would already have exited when the server signalled "scanner done".

And having a different flow for debug mode sounds wrong to me, too. I believe in the above the `elsif (main::SCANNER)` would hardly ever be hit, as nobody is going to disable debug logging for the scanner manually.